papermill/execute.py

```diff
@@ -182,9 +182,11 @@ def execute_notebook(notebook,
     nb.metadata.papermill['exception'] = any(
         [cell.metadata.papermill.get('exception') for cell in nb.cells])
 
-    # Write final Notebook to disk.
-    write_ipynb(nb, output)
     raise_for_execution_errors(nb, output)
+    if output is not None:
+        # Write final Notebook to disk.
+        write_ipynb(nb, output)
+    return nbformat.writes(nb)
```

Comment:
This will write the entire notebook to stdout, correct? We definitely don't want that to always happen.

Reply:
@rgbkrk Thanks, but what if I want to deal with output again? The only way is to read output file for now. Is there a way to drop file I/O step?

Reply:
@rgbkrk This use case is in Python API, not in the command line. :)

Reply:
Oh silly me, I see!

Reply:
In that case, it would be much better to return back the notebook object instead of the raw string? That allows you to strip out exactly what you need without having to deserialize the whole thing again. It should make testing the return much easier as well.

```
if output is not None:
  # Write final Notebook to disk.
  write_ipynb(nb, output)

return nb
```

Reply:
@rgbkrk Yeah, dealing with notebook object is much easier, I made the change. Please review, thanks!